src/components/react-hook-form/ExpandingTextField.tsx

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import { useState } from 'react';
+import { Box, InputAdornment, Theme } from '@mui/material';
+import { FunctionComponent } from 'react';
+import { useFormContext, useWatch } from 'react-hook-form';
+import { TextInput } from '../..';
+
+interface ExpandingTextFieldProps {
+    name: string;
+    maxCharactersNumber?: number;
+    rows?: number;
+    minRows?: number;
+    sx?: any;
+    label?: string;
+    textFieldFormProps?: any;
+    otherTexFieldProps?: any;
+}
+const ExpandingTextField: FunctionComponent<ExpandingTextFieldProps> = ({
+    name,
+    maxCharactersNumber,
+    rows,
+    minRows = 1,
+    sx,
+    label,
+    textFieldFormProps,
+    otherTexFieldProps,
+}) => {
+    const [isFocused, setIsFocused] = useState(false);
+    const { control } = useFormContext();
+    const descriptionWatch = useWatch({
+        name: name,
+        control,
+    });
+    const handleFocus = () => {
+        setIsFocused(true);
+    };
+
+    const handleBlur = () => {
+        setIsFocused(false);
+    };
+
+    const maxCounter = maxCharactersNumber ?? 500;
+    const isOverTheLimit = descriptionWatch?.length > maxCounter;
+    const descriptionLength = descriptionWatch?.length ?? 0;
+    const descriptionCounter = descriptionLength + '/' + maxCounter;
+
+    const rowsToDisplay = isFocused ? rows : minRows;
+
+    const formProps = {
+        size: 'medium',
```

Comment:
why not the default to `'small'`defined in TextField ?

Reply:
by default in gridexplore , the value medium is used, so to avoid passing this param every where i made it a default value